servant-client/src/Servant/Common/Req.hs

```diff
@@ -148,11 +148,11 @@ performRequest reqMethod req isWantedStatus reqHost = do
                                }
 
   eResponse <- liftIO $ __withGlobalManager $ \ manager ->
-    catchHttpException $
-    Client.httpLbs request manager
+    catchConnectionError $
+      Client.httpLbs request manager
   case eResponse of
     Left err ->
-      left $ ConnectionError err
+      left $ ConnectionError (show err)
```

Comment:
Do we have to lose the underlying exception here? Can we just pass this thing along as a SomeException existential? Then the user can recover the exception, or just throw/show it.